pysph/tools/binder.py

```
@@ -0,0 +1,208 @@
+    ]
+    cell3_src = [
+        "viewer = Viewer" + viewer_type + "(cwd)\n",
+        "viewer.interactive_plot(" + str(config_dict) + ")"
+    ]
+
+    nb = nbf.v4.new_notebook()
+
+    nb.cells = [
+        nbf.v4.new_code_cell(source=cell1_src),
+        nbf.v4.new_code_cell(source=cell2_src),
+        nbf.v4.new_code_cell(source=cell3_src)
+    ]
+
+    nbf.write(nb, path+'/'+sim_name+'.ipynb')
+    return
+
+
+def find_sim_dirs(path, sim_paths_list=[]):
+    '''
+    Finds all the directories in a given directory that
+    contain pysph output files.
+    '''
+    path = os.path.abspath(path)
+    sim_files = get_files(path)
+    if len(sim_files) != 0:
+        sim_paths_list.append(path)
+    elif len(sim_files) == 0:
+        files = os.listdir(path)
+        files = [f for f in files if os.path.isdir(f)]
+        files = [os.path.abspath(f) for f in files if not f.startswith('.')]
+        for f in files:
+            sim_paths_list = find_sim_dirs(f, sim_paths_list)
+
+    return sim_paths_list
+
+
+def find_dir_size(path):
+    '''
+    Finds the size of a given directory.
+    '''
+    total_size = 0
+    for dir_path, dir_names, file_names in os.walk(path):
+        for f in file_names:
+            fp = os.path.join(dir_path, f)
+            if not os.path.islink(fp):
+                total_size += os.path.getsize(fp)
+
+    return total_size
+
+
+def make_binder(path):
+
+    src_path = os.path.abspath(path)
+
+    sim_paths_list = find_sim_dirs(src_path)
+
+    for path in sim_paths_list:
+        sim_name = os.path.split(path)[1]
+        make_notebook(path, sim_name)
+        if len(sim_paths_list) == 1 and sim_paths_list[0] == src_path:
+            files = os.listdir(src_path)
+            files = [src_path+'/'+f for f in files]
+            mkdir(src_path+'/'+sim_name)
+            could_not_copy = []
+            for f in files:
+                try:
+                    copy(f, src_path+'/'+sim_name)
+                except:
```

Do not use catch-all except clauses as it will ignore any errors which can be dangerous.  Just catch the ones you need.

---

I've updated this block to where it captures all exceptions and prints them along with the files that could not be copied.